connector/src/main/java/org/geysermc/connector/network/translators/world/block/entity/SignBlockEntityTranslator.java

```diff
@@ -101,27 +102,34 @@ public void translateTag(NbtMapBuilder builder, CompoundTag tag, int blockState)
             String signLine = getOrDefault(tag.getValue().get("Text" + currentLine), "");
             signLine = MessageTranslator.convertMessageLenient(signLine);
 
-            // Trim any trailing formatting codes
-            if (signLine.length() > 2 && signLine.toCharArray()[signLine.length() - 2] == '\u00a7') {
-                signLine = signLine.substring(0, signLine.length() - 2);
-            }
-
             // Check the character width on the sign to ensure there is no overflow that is usually hidden
             // to Java Edition clients but will appear to Bedrock clients
             int signWidth = 0;
             StringBuilder finalSignLine = new StringBuilder();
+            boolean previousCharacterWasSymbol = false; // Color changes do not count for maximum width
```

Comment:
This should probably be renamed to something like `previousCharacterWasFormatting`. To make it clear that its not just any symbol (I know that's mentioned in the comment).